src/__tests__/patients/visits/AddVisitModal.test.tsx

```diff
@@ -28,77 +31,68 @@ describe('Add Visit Modal', () => {
   } as Patient
 
   const onCloseSpy = jest.fn()
-  const setup = () => {
+  const render = () => {
     jest.spyOn(PatientRepository, 'find').mockResolvedValue(patient)
     jest.spyOn(PatientRepository, 'saveOrUpdate')
     const history = createMemoryHistory()
-    const wrapper = mount(
-      <Router history={history}>
-        <AddVisitModal show onCloseButtonClick={onCloseSpy} patientId={patient.id} />
-      </Router>,
-    )
 
-    wrapper.update()
+    function Wrapper({ children }: WrapperProps) {
+      return <Router history={history}>{children}</Router>
+    }
 
-    return { wrapper: wrapper as ReactWrapper }
-  }
+    const results = rtlRender(
+      <AddVisitModal show onCloseButtonClick={onCloseSpy} patientId={patient.id} />,
+      {
+        wrapper: Wrapper,
+      },
+    )
 
-  it('should render a modal', () => {
-    const { wrapper } = setup()
+    return results
+  }
 
-    const modal = wrapper.find(Modal)
+  it('should render a modal and within a form', () => {
+    render()
 
-    expect(modal).toHaveLength(1)
+    expect(screen.getByRole('dialog').querySelector('form')).toBeInTheDocument()
+  })
 
-    const successButton = modal.prop('successButton')
-    const cancelButton = modal.prop('closeButton')
-    expect(modal.prop('title')).toEqual('patient.visits.new')
-    expect(successButton?.children).toEqual('patient.visits.new')
-    expect(successButton?.icon).toEqual('add')
-    expect(cancelButton?.children).toEqual('actions.cancel')
+  it('should call the on close function when the cancel button is clicked', () => {
+    render()
+    userEvent.click(
+      screen.getByRole('button', {
+        name: /close/i,
+      }),
+    )
+    expect(onCloseSpy).toHaveBeenCalledTimes(1)
   })
 
-  it('should render the visit form', () => {
-    const { wrapper } = setup()
+  it('should save the visit when the save button is clicked', () => {
+    render()
+    const testPatient = patient.visits[0]
+    const modal = screen.getByRole('dialog')
 
-    const addVisitModal = wrapper.find(AddVisitModal)
-    expect(addVisitModal).toHaveLength(1)
-  })
+    /* Date Pickers */
+    const modalDatePickerWrappers = modal.querySelectorAll('.react-datepicker__input-container')
+    const startDateInput = modalDatePickerWrappers[0].querySelector('input') as HTMLInputElement
+    const endDateInput = modalDatePickerWrappers[1].querySelector('input') as HTMLInputElement
```

Comment:
Curious why do you need to add as HTMLInputElement at the end of the datePickers?

Reply:
@emma-r-slight 
Good Question!
TypeScript complains that it could be null, this is a hacky way of telling TypeScript to be quiet and that this will be an input element. This is needed here because I am using querySelector from modal since I could not use a better query unfortunately :cry:, this would not happen if you use the other queries like `getByLabelText` :tada: